### PR TITLE
fix(permit): allow third parties to use permit

### PR DESCRIPTION
**Pull request by PierrickGT**
Issue: https://github.com/code-423n4/2023-07-pooltogether-findings/issues/113

### CODE REVIEW

File: lcov.info [hunk 1]
```diff
@@ -1,18 +1,18 @@
 TN:
 SF:src/Vault.sol
```
Comment (asselstine) [inline lcov.info:2]: ✂️  lcov from git

File: src/Vault.sol [hunk 2]
```diff
@@ -505,22 +498,22 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
...
     address _receiver,
     uint256 _deadline,
     uint8 _v,
     bytes32 _r,
     bytes32 _s
   ) external returns (uint256) {
-    _permit(IERC20Permit(asset()), msg.sender, address(this), _assets, _deadline, _v, _r, _s);
-    return deposit(_assets, _receiver);
+    _permit(IERC20Permit(asset()), _owner, address(this), _assets, _deadline, _v, _r, _s);
+    return _depositAssets(_assets, _owner, _receiver);
```
Comment (asselstine) [inline src/Vault.sol]: The _receiver isn't checked as part of the signature, which means that anyone can pass an arbitrary `_receiver`.

Comment (asselstine) [reply src/Vault.sol]: Without a custom signature, I think the only way to do this is have receiver = owner.

Thoughts?

File: src/Vault.sol [hunk 3]
```diff
@@ -546,13 +540,14 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
...
+    address _owner,
     uint256 _deadline,
     uint8 _v,
     bytes32 _r,
     bytes32 _s
   ) external returns (uint256) {
-    _permit(IERC20Permit(asset()), msg.sender, address(this), _assets, _deadline, _v, _r, _s);
-    return _sponsor(_assets);
+    _permit(IERC20Permit(asset()), _owner, address(this), _assets, _deadline, _v, _r, _s);
+    return _sponsor(_assets, _owner);
```
Comment (asselstine) [inline src/Vault.sol:550]: This makes sense, because the owner is the one signing off their own sponsorship.

**Review by asselstine — CHANGES_REQUESTED** [review]

File: src/Vault.sol (hunk 2 above)
Comment (PierrickGT) [reply src/Vault.sol]: > The _receiver isn't checked as part of the signature, which means that anyone can pass an arbitrary `_receiver`.

Ah yes, the signature could be reused to transfer funds to an arbitrary receiver.
Yes, looks like we can't deposit on behalf when using a signature.
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/20/commits/8cabf072630012a1543a6a575800be12d9910214

**Review by asselstine — APPROVED** [review]

**Pull request merged**